bus-mapping/src/evm/opcodes.rs

```diff
@@ -346,12 +346,19 @@ pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<ExecStep, Er
 
     // Increase caller's nonce
     let caller_address = call.caller_address;
-    let nonce_prev = state.sdb.increase_nonce(&caller_address);
+    let nonce_prev = state.sdb.get_account(&caller_address).1.nonce.as_u64();
+    // Increase caller's nonce when the tx is not invalid
+    let nonce = if !(state.tx.invalid_tx == Some(true)) {
```

Comment:
!= ? 
or maybe if let is better for option check.

Reply:
Updated.